Model/app.js

```diff
@@ -32,59 +26,65 @@ app.controller('myController', ['$scope', ($scope) => {
         return toastr.error('This is not a json file');
       }
       if ($scope.fileNames.includes(file[i].name)) {
-        return toastr.error(`${file[i].name} has already been uploaded`, 'Error');
+        return toastr.error(`${file[i].name} has been uploaded`, 'Error');
       }
       $scope.documents = $scope.fileNames.push(file[i].name);
-      $scope.showUploaded = true;
-      $scope.$apply($scope.documents);
-      toastr.success(`${file[i].name} uploaded successfully`, 'Success');
-
-      const reader = new FileReader();
-      reader.onload = () => {
-        $scope.fileContent[file[i].name] = reader.result;
-        $scope.$apply();
-      };
-      reader.readAsText(file[i]);
+      $scope.$apply();
+      InvertedIndex.readFile(file[i], (e) => {
+        const data = JSON.parse(e.target.result);
+        const fileCheck = InvertedIndex.validateFile(data);
+        if (fileCheck.status) {
+          $scope.fileContent[file[i].name] = data;
+          $scope.$apply();
+          toastr.success(`${file[i].name} uploaded successfully`, 'Success');
+        } else {
+          return toastr.error(fileCheck.msg);
+        }
+      });
     }
   };
-
   $scope.createIndex = () => {
     const fileName = document.getElementById('selectFile').value;
-    let fileContent = $scope.fileContent[fileName];
-
-
+    const fileContent = $scope.fileContent[fileName];
+    if (fileName.length === 0) {
+      return toastr.error('Upload a file before you create index', 'Error');
+    }
     try {
+      // fileContent = JSON.parse(fileContent);
       invertedIndex.createIndex(fileName, fileContent);
     } catch (err) {
       toastr.error(err.message);
     }
-
     const index = invertedIndex.getIndex(fileName);
     $scope.indices[fileName] = index;
-
-    fileContent = JSON.parse(fileContent);
     $scope.fileContent[fileName] = fileContent;
     const length = fileContent.length;
```

Comment:
This would still achieve the same thing the `for` loop is achieving.
```
let i = 0;
fileContent.forEach( () => {
  temp.push(i);
  i += 1;
});
```